### PR TITLE
feat: validate all curves in UBAConfig constructor

### PR DESCRIPTION
**Pull request by james-a-morris**

### CODE REVIEW

File: src/UBAFeeCalculator/UBAFeeCalculator.test.ts [hunk 1]
```diff
@@ -47,7 +47,7 @@ describe("UBA Fee Calculations", () => {
           upperBound: { target: toBN("0"), threshold: toBN("0") },
         },
       },
-      { default: [] },
+      { default: [[toBN("0"), parseUnits("0", 18)]] },
```
Comment (nicholaspai) [inline src/UBAFeeCalculator/UBAFeeCalculator.test.ts:50]: a zero is a zero :)

```suggestion
      { default: [[toBN("0"), toBN("0")]] },
```

File: src/UBAFeeCalculator/UBAFeeConfig.ts [hunk 2]
```diff
@@ -61,6 +62,8 @@ class UBAConfig {
    * @param lpGammaFunction A record of piecewise functions for each chain that define the utilization fee to ensure that the bridge responds to periods of high utilization
    * @param incentivePoolAdjustment A DAO controlled variable to track any donations made to the incentivePool liquidity
    * @param ubaRewardMultiplier Used to scale rewards when a fee is larger than the incentive balance
+   * @param assertValidityOfFeeCurves Assert the validity of all fee curves. If true, this will throw an error if any of the fee curves are invalid. If false, this will not throw an error if any of the fee curves are invalid.
```
Comment (nicholaspai) [inline src/UBAFeeCalculator/UBAFeeConfig.ts:65]: Can you remove these comments?

Comment (nicholaspai) [inline src/UBAFeeCalculator/UBAFeeConfig.ts:65]:
```suggestion
```

File: src/UBAFeeCalculator/UBAFeeConfig.ts [hunk 3]
```diff
@@ -76,6 +79,30 @@ class UBAConfig {
...
+  private assertValidityOfAllFeeCurves(): void {
+    // Find all the fee curves that could possibiliy be used
+    // in the UBA fee calculation. Specifically, these are the
+    // balancing fee curve and the lp gamma function curve. The
+    // curves are available for all overrides as well as their
+    // default counterparts.
+    const allBalancingFlows = [
+      this.balancingFee.default,
+      ...Object.values(this.balancingFee.override ?? {}),
+      this.lpGammaFunction.default,
```
Comment (nicholaspai) [inline src/UBAFeeCalculator/UBAFeeConfig.ts:101]: Does the lpGammaFunction need to have the same requirements as the balancing fee? Specifically, does it need to meet the requirement that it passes the 0 y-value at some point?

**Review by nicholaspai — COMMENTED** [review]
A few nits then lgtm

File: src/UBAFeeCalculator/UBAFeeConfig.ts (hunk 3 above)
Comment (james-a-morris) [reply src/UBAFeeCalculator/UBAFeeConfig.ts:100]: I will check with Chase

**Pull request merged**